packages/plugins/extended-validation/test/one-of.spec.ts

```diff
@@ -204,7 +371,84 @@ describe('oneOf', () => {
           },
         ],
-      ])('%s', async (title, { document, variables, expectedError }) => {
+        [
+          'Invalid: More than one value is specific for deeply nested oneOf input type',
+          {
+            document: `query user($input: NestedOneOfFieldInput!) { nestedOneOf(input: $input) }`,
+            variables: {
+              input: {
+                field: {
+                  id: 1,
+                  username: 'test',
+                },
+              },
+            },
+            expectedError: 'Exactly one key must be specified for input type "UserUniqueCondition"',
+          },
+        ],
+        [
+          'Invalid: More than one value is specific for deeply nested oneOf input type',
+          {
+            document: `query user($input: DeeplyNestedOneOfFieldInput!) { deeplyNestedOneOf(input: $input) }`,
+            variables: {
+              input: {
+                field: {
+                  field: {
+                    id: 1,
+                    username: 'test',
+                  },
+                },
+              },
+            },
+            expectedError: 'Exactly one key must be specified for input type "UserUniqueCondition"',
+          },
+        ],
+        [
+          'Invalid: oneOf input list with more than one value specified on the oneOf type',
+          {
+            document: `query user($input: [UserUniqueCondition!]) { listOneOf(input: $input) }`,
+            variables: {
+              input: [
+                {
+                  id: 1,
+                  username: 'test',
+                },
+              ],
+            },
+            expectedError: 'Exactly one key must be specified for input type "UserUniqueCondition"',
+          },
+        ],
+        [
+          'Invalid: oneOf object field with incorrect object value',
+          {
+            document: DOCUMENT_WITH_WHOLE_INPUT,
+            variables: {
+              input: 1
+            },
+            expectedError: `Variable "$input" got invalid value 1; Expected type "UserUniqueCondition" to be an object.`,
+          },
+        ],
+        [
+          'Invalid: oneOf input list with incorrect list value',
+          {
+            document: `query user($input: [UserUniqueCondition!]) { listOneOf(input: $input) }`,
+            variables: {
+              input: 1
+            },
+            expectedError: `Variable "$input" got invalid value 1; Expected type "UserUniqueCondition" to be an object.`,
+          },
+        ],
+        [
+          'Invalid: oneOf input list with incorrect list value (variant with weird behavior)',
+          {
+            document: `query user($input: [UserUniqueCondition!]) { listOneOf(input: $input) }`,
+            variables: {
+              input: { a: 1 }
+            },
+            expectedError: `Variable "$input" got invalid value { a: 1 }; Field "a" is not defined by type "UserUniqueCondition".`,
+          },
+        ],
```

Comment:
This one is a bit weird. Does it seem like graphql is automatically wrapping an object in an array?